### PR TITLE
Add dot net blog post

### PR DESCRIPTION
**Pull request by mooreds**
This blog post shows how to use the .NET core client libraries to add a user to an application in FusionAuth.

### CODE REVIEW

**Comment by mooreds** [comment]
@pgeoly will review

File: site/_posts/2020-04-22-dot-net-command-line-client.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+This application has two main components. 
+
+The first is the .NET Core CLI tool. This will create a user by calling the FusionAuth APIs. 
+
+And then there is the FusionAuth identity server, which is a standalone application accessible at `http://localhost:9011`. This will be accessed by you, during configuration and also via API calls.
+
+## APIs and client libraries
+
+FusionAuth has a [full featured and well documented API](https://fusionauth.io/docs/v1/tech/apis/) which allows you to create and manage users, applications, and groups. Using this API allows you to extend FusionAuth for purposes never contemplated by the FusionAuth team. You can also integrate it with other data sources and pieces of software unknown to us. If your software can speak JSON and reach FusionAuth over a network connection, you can integrate.
```
Comment (pgeoly) [inline site/_posts/2020-04-22-dot-net-command-line-client.md:45]: Revise to: [full-featured and well-documented API]

File: site/_posts/2020-04-22-dot-net-command-line-client.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+{% include _image.html src="/assets/img/blogs/dot-net-command-line-client/create-application.png" alt="The application in FusionAuth after it has been created" class="img-fluid" figure=false %}
+
+Then go to the APIs section. We'll need to create an API key for our CLI client. Head to "Settings" and then to "API Keys" in the UI. Create a new key, and set the permissions. 
+
+Following the principle of least privilege, we'll allow POSTs to `/api/user` and `/api/user/registration` and nothing else. These privileges will let the owner of this key create users and user registrations.
+
+Note the key value (it looks something like `J9NXRWmkLVqt2hLc670s-i5iWzdPpgT_uXLuJcnMaFO`), as we'll need it later.
+
+## The command line client
```
Comment (pgeoly) [inline site/_posts/2020-04-22-dot-net-command-line-client.md:77]:
```suggestion
## The command-line client
```

File: site/_posts/2020-04-22-dot-net-command-line-client.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+	    registrationRequest.skipVerification = true;
+	    registrationRequest.registration = registration;
+            return client.Register(user.id, registrationRequest);
+        }
+    }
+}
+
+``` 
+
+I'm not going to review every line, but will highlight a few interesting points.
```
Comment (pgeoly) [inline site/_posts/2020-04-22-dot-net-command-line-client.md:202]:
```suggestion
I'm not going to review every line but will highlight a few interesting points.
```

File: site/_posts/2020-04-22-dot-net-command-line-client.md [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+...
+        private static readonly string apiKey = Environment.GetEnvironmentVariable("fusionauth_api_key");
+        private static readonly string fusionauthURL = "http://localhost:9011";
+
+	private static readonly string tenantId = "66636432-3932-3836-6630-656464383862";
+	private static readonly string applicationId = "4243b56f-0b45-4882-aa23-ac75eea22d22";
+...
+```
+
+We build the user request object first. This inclues basic information, such as the email and password. The password will be encrypted at rest using the tenant default encryption settings. If this weren't a tutorial, you'd also be connecting to FusionAuth over TLS, encrypting the password in transit.
```
Comment (pgeoly) [inline site/_posts/2020-04-22-dot-net-command-line-client.md:215]: Correct spelling in second sentence. inclues -> includes

File: site/_posts/2020-04-22-dot-net-command-line-client.md [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+layout: blog-post
+title: Creating a user in FusionAuth with a dot net cli
+description: How do you use the API in FusionAuth to create a user?
+author: Dan Moore
+image: blogs/whats-new-in-oauth-2-1/whats-new-with-oauth-2-1.png
+category: blog
+excerpt_separator: "<!--more-->"
+---
+
+In this post, we'll build a command line client for FusionAuth with C# and the .NET Core libraries. We'll set up FusionAuth and then add a user to an application from the command line interface (CLI) tool we've built.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2020-04-22-dot-net-command-line-client.md:11]: Per our discussion on hangouts, I've never seen command line hyphenated. :)

**Comment by mooreds** [comment]
@robotdan at your leisure, please create the correct FusionAuth images for this post. Aim is to publish next week, so no rush.

**Pull request merged**